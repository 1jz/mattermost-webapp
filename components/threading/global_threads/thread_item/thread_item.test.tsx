// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {ComponentProps} from 'react';
import {shallow} from 'enzyme';

import ThreadItem from './thread_item';

const users = [
    {
        url: 'test-url-1',
        username: 'johnny.depp',
        name: 'Johnny Depp',
    },
    {
        url: 'test-url-2',
        username: 'bilbo.baggins',
        name: 'Bilbo Baggins',
    },
    {
        url: 'test-url-3',
        username: 'michael.hall',
        name: 'Anthony Michael Hall',
    },
    {
        url: 'test-url-4',
        username: 'kathy.baker',
        name: 'Kathy Baker',
    },
    {
        url: 'test-url-5',
        username: 'vincent.price',
        name: 'Vincent Price',
    },
    {
        url: 'test-url-6',
        username: 'alan.arkin',
        name: 'Alan Arkin',
    },
];

describe('components/threading/global_threads/thread_item', () => {
    type Props = ComponentProps<typeof ThreadItem>;
    let actions: Props['actions'];

    beforeEach(() => {
        actions = {
            select: jest.fn(),
            follow: jest.fn(),
            unfollow: jest.fn(),
            openInChannel: jest.fn(),
            markRead: jest.fn(),
            markUnread: jest.fn(),
            save: jest.fn(),
            unsave: jest.fn(),
            copyLink: jest.fn(),
        };
    });

    test('should match snapshot', () => {
        const wrapper = shallow<typeof ThreadItem>(
            <ThreadItem
                name={'Johnny Depp'}
                teamName={'Enterprise Team'}
                previewText={'Lobortis phasellus feugiat vivamus facilisis ac suspendisse elit orci augue, metus hac libero cum diam accumsan magnis.'}
                participants={users}
                totalReplies={9}
                newReplies={2}
                newMentions={1}
                lastReplyAt={new Date('2020-09-29T02:30:15.701Z')}

                isFollowing={true}
                isSaved={false}
                isSelected={false}

                actions={actions}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
