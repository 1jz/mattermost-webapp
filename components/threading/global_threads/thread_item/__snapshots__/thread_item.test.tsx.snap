// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/threading/global_threads/thread_item should report total number of replies 1`] = `
<div
  className="ThreadItem"
  onClick={[MockFunction]}
  tabIndex={0}
>
  <h4>
    Johnny Depp
    <Badge
      className=""
      show={true}
    >
      Enterprise Team
    </Badge>
    <Connect(injectIntl(Timestamp))
      className="alt-hidden"
      units={
        Array [
          "now",
          "minute",
          "hour",
          "today-yesterday",
          "week",
          "month",
        ]
      }
      useTime={false}
      value={2020-09-29T02:30:15.701Z}
    />
    <span
      className="menu-anchor alt-visible"
    >
      <Memo(ThreadMenu)
        actions={
          Object {
            "copyLink": [MockFunction],
            "follow": [MockFunction],
            "markRead": [MockFunction],
            "markUnread": [MockFunction],
            "openInChannel": [MockFunction],
            "save": [MockFunction],
            "unfollow": [MockFunction],
            "unsave": [MockFunction],
          }
        }
        hasUnreads={false}
        isFollowing={true}
        isSaved={false}
      />
    </span>
  </h4>
  <p>
    Lobortis phasellus feugiat vivamus facilisis ac suspendisse elit orci augue, metus hac libero cum diam accumsan magnis.
  </p>
  <div
    className="activity"
  >
    <Memo(Avatars)
      size="sm"
      users={
        Array [
          Object {
            "name": "Johnny Depp",
            "url": "test-url-1",
            "username": "johnny.depp",
          },
          Object {
            "name": "Bilbo Baggins",
            "url": "test-url-2",
            "username": "bilbo.baggins",
          },
          Object {
            "name": "Anthony Michael Hall",
            "url": "test-url-3",
            "username": "michael.hall",
          },
          Object {
            "name": "Kathy Baker",
            "url": "test-url-4",
            "username": "kathy.baker",
          },
          Object {
            "name": "Vincent Price",
            "url": "test-url-5",
            "username": "vincent.price",
          },
          Object {
            "name": "Alan Arkin",
            "url": "test-url-6",
            "username": "alan.arkin",
          },
        ]
      }
    />
    <FormattedMessage
      defaultMessage="{totalReplies, plural, =0 {Reply} =1 {# reply} other {# replies}}"
      id="threading.numReplies"
      values={
        Object {
          "totalReplies": 9,
        }
      }
    />
  </div>
</div>
`;

exports[`components/threading/global_threads/thread_item should report unread mentions 1`] = `
<div
  className="ThreadItem has-indicator"
  onClick={[MockFunction]}
  tabIndex={0}
>
  <h4>
    <div
      className="indicator"
    >
      <div
        className="dot-mentions"
      >
        1
      </div>
    </div>
    Johnny Depp
    <Badge
      className=""
      show={true}
    >
      Enterprise Team
    </Badge>
    <Connect(injectIntl(Timestamp))
      className="alt-hidden"
      units={
        Array [
          "now",
          "minute",
          "hour",
          "today-yesterday",
          "week",
          "month",
        ]
      }
      useTime={false}
      value={2020-09-29T02:30:15.701Z}
    />
    <span
      className="menu-anchor alt-visible"
    >
      <Memo(ThreadMenu)
        actions={
          Object {
            "copyLink": [MockFunction],
            "follow": [MockFunction],
            "markRead": [MockFunction],
            "markUnread": [MockFunction],
            "openInChannel": [MockFunction],
            "save": [MockFunction],
            "unfollow": [MockFunction],
            "unsave": [MockFunction],
          }
        }
        hasUnreads={true}
        isFollowing={true}
        isSaved={false}
      />
    </span>
  </h4>
  <p>
    Lobortis phasellus feugiat vivamus facilisis ac suspendisse elit orci augue, metus hac libero cum diam accumsan magnis.
  </p>
  <div
    className="activity"
  >
    <Memo(Avatars)
      size="sm"
      users={
        Array [
          Object {
            "name": "Johnny Depp",
            "url": "test-url-1",
            "username": "johnny.depp",
          },
          Object {
            "name": "Bilbo Baggins",
            "url": "test-url-2",
            "username": "bilbo.baggins",
          },
          Object {
            "name": "Anthony Michael Hall",
            "url": "test-url-3",
            "username": "michael.hall",
          },
          Object {
            "name": "Kathy Baker",
            "url": "test-url-4",
            "username": "kathy.baker",
          },
          Object {
            "name": "Vincent Price",
            "url": "test-url-5",
            "username": "vincent.price",
          },
          Object {
            "name": "Alan Arkin",
            "url": "test-url-6",
            "username": "alan.arkin",
          },
        ]
      }
    />
    <FormattedMessage
      defaultMessage="{newReplies, plural, =1 {# new reply} other {# new replies}}"
      id="threading.numNewReplies"
      values={
        Object {
          "newReplies": 2,
        }
      }
    />
  </div>
</div>
`;

exports[`components/threading/global_threads/thread_item should report unread messages 1`] = `
<div
  className="ThreadItem has-indicator"
  onClick={[MockFunction]}
  tabIndex={0}
>
  <h4>
    <div
      className="indicator"
    >
      <div
        className="dot-unreads"
      />
    </div>
    Johnny Depp
    <Badge
      className=""
      show={true}
    >
      Enterprise Team
    </Badge>
    <Connect(injectIntl(Timestamp))
      className="alt-hidden"
      units={
        Array [
          "now",
          "minute",
          "hour",
          "today-yesterday",
          "week",
          "month",
        ]
      }
      useTime={false}
      value={2020-09-29T02:30:15.701Z}
    />
    <span
      className="menu-anchor alt-visible"
    >
      <Memo(ThreadMenu)
        actions={
          Object {
            "copyLink": [MockFunction],
            "follow": [MockFunction],
            "markRead": [MockFunction],
            "markUnread": [MockFunction],
            "openInChannel": [MockFunction],
            "save": [MockFunction],
            "unfollow": [MockFunction],
            "unsave": [MockFunction],
          }
        }
        hasUnreads={true}
        isFollowing={true}
        isSaved={false}
      />
    </span>
  </h4>
  <p>
    Lobortis phasellus feugiat vivamus facilisis ac suspendisse elit orci augue, metus hac libero cum diam accumsan magnis.
  </p>
  <div
    className="activity"
  >
    <Memo(Avatars)
      size="sm"
      users={
        Array [
          Object {
            "name": "Johnny Depp",
            "url": "test-url-1",
            "username": "johnny.depp",
          },
          Object {
            "name": "Bilbo Baggins",
            "url": "test-url-2",
            "username": "bilbo.baggins",
          },
          Object {
            "name": "Anthony Michael Hall",
            "url": "test-url-3",
            "username": "michael.hall",
          },
          Object {
            "name": "Kathy Baker",
            "url": "test-url-4",
            "username": "kathy.baker",
          },
          Object {
            "name": "Vincent Price",
            "url": "test-url-5",
            "username": "vincent.price",
          },
          Object {
            "name": "Alan Arkin",
            "url": "test-url-6",
            "username": "alan.arkin",
          },
        ]
      }
    />
    <FormattedMessage
      defaultMessage="{newReplies, plural, =1 {# new reply} other {# new replies}}"
      id="threading.numNewReplies"
      values={
        Object {
          "newReplies": 2,
        }
      }
    />
  </div>
</div>
`;
